### PR TITLE
[WIP] FRP Transition

### PR DESCRIPTION
**Pull request by ashfurrow**
Unit tests are now compiling and completing, with twenty failures. I'll be looking at those next.

### CODE REVIEW

File: Kiosk/Auction Listings/ListingsViewModel.swift [hunk 1]
```diff
@@ -17,7 +17,7 @@ protocol ListingsViewModelType {
     var updatedContentsSignal: Observable<NSDate> { get }
 
     var scheduleOnBackground: (signal: Observable<AnyObject>) -> Observable<AnyObject> { get }
-    var scheduleOnForeground: (signal: Observable<[SaleArtwork]>) -> Observable<[SaleArtwork]> { get }
+    var scheduleOnForeground:  (signal: Observable<[SaleArtwork]>) -> Observable<[SaleArtwork]> { get }
```
Comment (orta) [inline Kiosk/Auction Listings/ListingsViewModel.swift]: you don't want this space

File: KioskTests/Models/SaleArtworkTests.swift [hunk 2]
```diff
@@ -79,17 +82,15 @@ class SaleArtworkTests: QuickSpec {
...
                 }
 
                 it("updates reserve status when reserve status changes") {
 
                     var reserveStatus = ""
                     saleArtwork.viewModel.numberOfBidsWithReserveSignal.subscribeNext { (newReserveStatus) -> Void in
-                        reserveStatus = newReserveStatus as! String
-                    }
+                        reserveStatus = newReserveStatus
+                    }.addDisposableTo(disposeBag)
```
Comment (orta) [inline KioskTests/Models/SaleArtworkTests.swift]: wonder if this could be "automated" at Quick or Nimble level?

**Comment by orta** [comment]
Cool, no surprises here :+1:

File: KioskTests/Models/SaleArtworkTests.swift (hunk 2 above)
Comment (ashfurrow) [inline KioskTests/Models/SaleArtworkTests.swift]: Yeah that's my hope. Along with the matchers, it'd make a nice tool!

**Pull request merged**